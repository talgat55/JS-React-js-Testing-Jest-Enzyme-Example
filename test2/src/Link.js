import React from 'react';
import {string} from 'prop-types';

class Link extends React.Component {

    handleClick()  {
        alert( 'Кликнули по ссылке!' ) ;
    };

render(){
    const {title, url} = this.props;

    return <a href={url} onClick={this.handleClick}>{title}</a>;
}};
Link.propTypes = {
    title: string.isRequired,
    url: string.isRequired
};
export default Link;