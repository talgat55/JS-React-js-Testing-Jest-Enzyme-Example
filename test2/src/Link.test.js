import React from 'react';
import Enzyme,{shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from "enzyme-to-json";
import Link from './Link';

Enzyme.configure({adapter: new Adapter()});
describe('Test Link Component', () =>{
    it('should handle the click event', () => {
        window.alert = jest.fn();
       const output = shallow(<Link title="Title" url="url" />);
        output.simulate('click');

       expect(window.alert).toHaveBeenCalledWith('click on link');
    });
});
it('should handle the click event', () => {
    window.alert = jest.fn();
    const output = shallow(
        <Link title="mockTitle" url="mockUrl" />
    );
    output.simulate('click');
    expect(window.alert).toHaveBeenCalledWith('Кликнули по ссылке!');
});