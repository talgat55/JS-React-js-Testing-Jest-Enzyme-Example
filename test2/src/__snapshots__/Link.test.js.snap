// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Link Component should render correctly 1`] = `
<a
  href="url"
>
  Title
</a>
`;
